src/main/java/org/apache/commons/io/input/CharSequenceInputStream.java

```diff
@@ -71,17 +72,21 @@ public CharSequenceInputStream(final CharSequence cs, final Charset charset) {
      * @throws IllegalArgumentException if the buffer is not large enough to hold a complete character.
      */
     public CharSequenceInputStream(final CharSequence cs, final Charset charset, final int bufferSize) {
+        Objects.requireNonNull(cs);
+        if (bufferSize < 1) {
+            throw new IllegalArgumentException("Buffer size must be >= 1");
+        }
+        // TODO: bufferSize is unused
```

Comment:
Will change this to pass it to `ReaderInputStream` and use it as inital `encoderOut` size there.